libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -698,6 +728,54 @@ protected async Task<RecognizerResult> OnRecognize(ActionContext actionContext,
             };
         }
 
+        protected virtual void EnsureDependenciesInstalled()
+        {
+            lock (this)
```

Comment:
Is there an option other than lock(this) that could be used? Many folks consider locking like that to be problematic, as it's easy for external sources to lock and end up in a deadlock. For a good discussion of this, Jon Skeet has a lengthy discussion: 
https://jonskeet.uk/csharp/threads/lockchoice.html